### PR TITLE
Backwards-compatible column formatter

### PR DESCRIPTION
**Pull request by davidkokkelink**
This PR adds a number & date formatting helpers, that are designed to be backwards-compatible to the `columnFormatter` that used to be bundled in `chart-core` for [date](https://github.com/datawrapper/chart-core/blob/master/lib/dw/dataset/columnTypes/date.js#L474-L519) and [number](https://github.com/datawrapper/chart-core/blob/master/lib/dw/dataset/columnTypes/number.js#L113-L148).

The old implementations used `Globalize`. Over time, we have gradually replaced the usage of these `columnFormatters` in visualization plugins with `dayjs` and `numeral`. However, there are certain visualizations have stored the _output_ of the legacy formatter functions in chart metadata (namely bar & column charts to save which bar/columns were highlighted). We also still use them for the display of the data table in step 2.

This creates the need for a backwards-compatible number formatter - something that given the same parameters, results in the same formatted output, but using `dayjs` and `numeral` instead of `Globalize`. That's what this PR adds. I added tests for English - there might be inconsistencies in other locales between `dayjs` and `Globalize` formatting, in which case it might not be perfectly backwards-compatible. Still, I think this is good enough to allow us to get rid of the `Globalize` dependency everywhere.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -27,6 +27,7 @@
         "url": "git+ssh://git@github.com/datawrapper/shared.git"
     },
     "devDependencies": {
+        "@datawrapper/chart-core": "8.30",
```
Comment (gka) [inline package.json:30]: I don't see `chart-core` being imported anywhere in this PR so perhaps we can remove it from the dev dependencies.

**Review by gka — CHANGES_REQUESTED** [review]
looks good, otherwise

**Comment by gka** [comment]
just an idea what came up while reviewing https://github.com/datawrapper/plugin-d3-bars/pull/89;

how about we add another helper function `columnFormatter(column, chart)`:

```js
import dateColumnFormatter from './dateColumnFormatter';
import numberColumnFormatter from './numberColumnFormatter';
import get from './get';

const identity = d => d;

export function columnFormatter(column, metadata, axis) {
    return column.type() === 'date'
        ? dateColumnFormatter(column.type(true))
        : column.type() === 'number'
        ? numberColumnFormatter(get(metadata, `data.column-format.${axis}`, {}))
        : identity;
}
```

in `d3-bars` (and other plugins) we would then use it like this

```js
const fmt = columnFormatter(
    colorBy,
    this.chart().getMetadata(),
    his.axes().colors || this.axes().labels
);
const labelFormatter = columnFormatter(
    vis.axes(true).labels,
    vis.chart().getMetadata(),
    vis.axes().labels
);
```

File: package.json (hunk 1 above)
Comment (davidkokkelink) [reply package.json:30]: you're right :+1:

**Comment by davidkokkelink** [comment]
makes sense, I added your `columnFormatter` suggestion

**Review by gka — APPROVED** [review]

**Pull request merged**